lib/improver/units.py

```diff
@@ -62,10 +62,55 @@
     "projection_y_coordinate": {"unit": "m"},
     "percentile": {"unit": "%"},
     "probability": {"unit": "1"},
+    "realization": {"unit": "1", "dtype": np.int32},
+    "height": {"unit": "m"},
+    "velocity": {"unit": "m s-1"},
     # standard diagnostics and suitable substrings (alphabetised for clarity)
+    "air_temperature_lapse_rate": {"unit": "K m-1"},
+    "cloud": {"unit": "1"},
     "fall_rate": {"unit": "m s-1"},
+    "falling_snow_level": {"unit": "m"},
+    "humidity": {"unit": "1"},
     "lapse_rate": {"unit": "K m-1"},
+    "orographic_enhancement": {"unit": "m s-1"},
     "precipitation_rate": {"unit": "m s-1"},
+    "pressure": {"unit": "Pa"},
     "temperature": {"unit": "K"},
-    "thickness": {"unit": "m"}
+    "thickness": {"unit": "m"},
+    "ultraviolet_flux": {"unit": "W m-2"},
+    "ultraviolet_index": {"unit": "1"},
+    "visibility_in_air": {"unit": "m"},
+    "weather_code": {"unit": "1", "dtype": np.int32},
+    "wind_from_direction": {"unit": "degrees"},
+    "wind_speed": {"unit": "m s-1"},
+    "wind_to_direction": {"unit": "degrees"},
+    # ancillary diagnostics
+    "alphas": {"unit": "1"},
+    "grid_with_halo": {"unit": "no_unit"},
+    "emos_coefficients": {"unit": "1"},
+    "surface_altitude": {"unit": "m"},
+    "land_binary_mask": {"unit": "1", "dtype": np.int32},
```

Comment:
Based on the notebook where you explain the inputs and outputs to the plugin that generates this, I think I need to understand the use case a little better.  The idea of processing a field, outputting something with the same name but a different interpretation and units doesn't sit well with me.

Reply:
I have renamed the float32 cube going into the generate-landmask CLI test as land_fraction. This requires an additional rename step in the CorrectLandSeaMask plugin just to ensure that the output has the expected name for a binary land mask field, which is what it produces.